src/adapters/store.js

```diff
@@ -2,112 +2,170 @@ import nconf from '@qwant/nconf-getter'
 import Error from '../adapters/error'
 import {version} from '../../config/constants.yml'
 import ExtendedString from "../libs/string";
+import LocalStore from "../libs/local_store"
+import MasqStore from "../libs/masq"
 
-let moduleConfig = nconf.get().store
+export default class Store {
 
-const AbStore = require(`../libs/${moduleConfig.name}`)
-const abstractStore = new AbStore(moduleConfig.endpoint)
+  constructor() {
+    // get store from window if already initialized
+    if (window.__store) {
+      return window.__store
+    }
+    // if store not initialized, use this
+    window.__store = this
 
-function Store() {
-  this.isRegisterd = false
-  if(!window.__existingStore) {
-    window.__existingStore = true
-    listen('store_poi', (poi) => {
-      this.add(poi)
-    })
-    listen('del_poi', (poi) => {
-      this.del(poi)
-    })
-    listen('store_center', (loc) => {
-      this.setLastLocation(loc)
-    })
-    listen('store_clear', () => {
-      this.clear()
-    })
+    // init stores
+    this.localStore = new LocalStore()
+    this.loggedIn = false
+    this.abstractStoreStr = 'local_store'
+    this.abstractStore = this.localStore
+    this.masqConfig = nconf.get().masq
+    if (this.masqConfig.enabled) {
+      this.masqStore = new MasqStore(this.masqConfig)
+      if (this.masqStore.isLoggedIn()) {
+        this.abstractStore = this.masqStore
+      }
+    }
```

Comment:
else new localStore 
should be more acceptable, here you instanciate local store, and you instanciate masqStore